### DISCUSSION TITLE
Add config option to define base URLs for GraphQL view queries

### DISCUSSION THREAD

**Discussion by el-schneider** [aerni/statamic-advanced-seo#89] [Ideas]
Querying e.g. `seo > view > head` uses the root URL of the statamic application for generating canonical links, which often doesn't make sense for a headless app. The favicon on the other hand seems to be a relative URL, which also breaks on a headless setup. I think it would be nice if there could be config options for `app_url` or `frontend_url` and `asset_url`, although Statamic only seems to be using the corresponding ENV vars `APP_URL` and `ASSET_URL` for CLI tools.

**Comment by aerni** [comment]
I believe there was a reason I didn't add this feature. But I can't quite remember why. I will look into it.

**Comment by aerni** [comment]
I took a look at this and there are a couple of things going on.

I can add a `baseUrl` argument that allows you to override Statamic's `APP_URL`. This argument is already used by some of the computed fields like `canonical` and `hreflang`. It would allow changing the URL of the `breadcrumbs`, `canonical`, `hreflang`, and `schema` fields in the rendered view. It could be used like this:

```graphql
query MyQuery {
  seoMeta(id: "home") {
    view {
      head(baseUrl: "https://foo.bar")
    }
  }
}
```

However, this won't have an effect on the URLs of asset fields like `og_image` and `favicon`. Those URLs are handled by Statamic's assets fieldtype and glide tag. You could change the URL of your asset container, which would work for the favicon. The social images, however, are using glide. And the URL of glide is coupled to the `APP_URL` and can't be changed.

To summarize, adding the `baseUrl` argument would get you half the way. Any thoughts?

**Comment by aerni** [comment]
As a side note, I actually already added a comment on the `renderedViews` type to only use it when Statamic's URLs work for you.

<img width="342" alt="CleanShot 2023-08-01 at 20 13 15@2x" src="https://github.com/aerni/statamic-advanced-seo/assets/23167701/a739cb56-8a23-4f3b-9f46-335e787fae9a">
